### PR TITLE
rename ligand chains

### PR DESCRIPTION
**Pull request by tomcastigl**
## Description
Running `plinder_eval` on multi-ligand systems ends up with OpenStructure complaining that the different ligands have the same chain name 
```
2024-11-07 08:59:20,580 | plinder.eval.docking.write_scores:159 | ERROR : get_scores: A residue number 1 already exists in chain 00001_Molecule
```

renaming the ligand chains fixes this!

### CODE REVIEW

File: src/plinder/eval/docking/utils.py [hunk 1]
```diff
@@ -85,10 +85,17 @@ def from_files(
...
-            for ligand_sdf_file in ligand_files
-        ]
+        ligand_views = []
+        for i, ligand_file in enumerate(ligand_files):
+            ligand_entity = io.LoadEntity(str(ligand_file), format="sdf")
+
+            # rename ligand chain to have different chain names for each ligand
+            # this is necessary for ost to not complain about duplicate chain names
+            editor = ligand_entity.EditXCS()
+            editor.RenameChain(list(ligand_entity.chains)[0], f"LIG_{i}")
```
Comment (OleinikovasV) [inline src/plinder/eval/docking/utils.py:95]: not sure about this chain label convention - as it is normally a single letter - I suppose a lowercase letter might be sufficient or just a number might also work. But as this is only for scoring - also happy with whatever works!
WDYT @Ninjani ?

Comment (tomcastigl) [reply src/plinder/eval/docking/utils.py:95]: yeah that was just me using the first thing that came to my mind 😅 happy to use a better convention!

Comment (Ninjani) [reply src/plinder/eval/docking/utils.py:95]: I think it's fine as we don't save the molecules anywhere!

File: src/plinder/eval/docking/utils.py [hunk 2]
```diff
@@ -420,40 +427,49 @@ def get_average_posebusters(self) -> dict[str, list[Any]]:
...
-            num_model_ligands=self.model.num_ligands,
-            num_reference_proteins=self.reference.num_proteins,
-            num_model_proteins=self.model.num_proteins,
-            fraction_reference_ligands_mapped=self.num_mapped_reference_ligands
-            / self.reference.num_ligands,
-            fraction_model_ligands_mapped=self.num_mapped_model_ligands
-            / self.model.num_ligands,
-        )
-        score_list = ["lddt_pli", "lddt_lp", "bisy_rmsd"]
-        for score_name in score_list:
```
Comment (OleinikovasV) [inline src/plinder/eval/docking/utils.py:437]: @Ninjani - as a result of this change we are no longer getting:
`bisy_rmsd_wave` , `bisy_rmsd_ave`, `lddt_pli_wave`, etc. scores.

What is the desired behaviour. We should decide and then update:
1) the docs:
- here: https://github.com/plinder-org/plinder/blob/main/docs/evaluation.md, 
- and here: https://github.com/plinder-org/plinder/blob/main/docs/dataset.md#linked-systems-links

2) make_plots code (multiple functions):
https://github.com/plinder-org/plinder/blob/main/src/plinder/eval/docking/make_plots.py

File: src/plinder/eval/docking/make_plots.py [hunk 3]
```diff
@@ -43,7 +43,7 @@ class EvaluationResults:
     def from_scores_and_data_files(
         cls, score_file: Path, data_file: Path, output_dir: Path, top_n: int = 10
     ) -> "EvaluationResults":
-        scores_df = pd.read_parquet(score_file)
+        scores_df = pd.read_parquet(score_file).drop_duplicates("system_id")
```
Comment (OleinikovasV) [inline src/plinder/eval/docking/make_plots.py:46]: this is to plot aggregated data with only one value per system

**Review by Ninjani — APPROVED** [review]

**Pull request merged**